keps/sig-cloud-provider/20200928-extend-konnectivity-for-both-directions.md

```diff
@@ -0,0 +1,194 @@
+---
+title: Out-of-Tree Credential Providers
+authors:
+  - "@irozzo-1A"
+  - "@youssefazrak"
+owning-sig: sig-cloud-provider
+reviewers:
+  - "@cheftako"
+approvers: TBD
+editor: TBD
+creation-date: 2020-09-28
+last-updated: 2020-09-28
+status: draft
+---
+
+# Extending Apiserver Network Proxy to handle traffic originated from Node network
+
+## Table of Contents
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Definitions](#definitions)
+- [Proposal](#proposal)
+  - [Traffic flow](#traffic-flow)
+  - [Agent additional flags](#agent-additional-flags)
+  - [Handling the Traffic from the Pods to the Agent](#handling-the-traffic-from-the-pods-to-the-agent)
+  - [Handling the Traffic from the Kubelet to the Agent](#handling-the-traffic-from-the-kubelet-to-the-agent)
+  - [Deployment Model](#deployment-model)
+  - [Listening Interface for Konnectivity agent](#listening-interface-for-konnectivity-agent)
+  - [Authentication](#authentication)
+  - [Risks and Mitigations](#risks-and-mitigations)
+    - [Allow list](#allow-list)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Infrastructure Needed](#infrastructure-needed)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+The goal of this proposal is to allow traffic to flow from the Node Network to the Master Network.
```

Comment:
"Cluster network to the Control Plane network"

Reply:
We initially used that terminology, but we changed it later to be consistent with the definitions below. Of course we are open to change if you think it makes it more clear.